### PR TITLE
fix(interchain-token-service): update error handling for token id config

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-6985](https://axelarnetwork.atlassian.net/browse/AXE-6985)

[AXE-6985]: https://axelarnetwork.atlassian.net/browse/AXE-6985?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/interchain-token-service/src/contract.rs [hunk 1]
```diff
@@ -161,27 +161,16 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
...
-    ///
-    /// # Returns
-    /// * `TokenManagerType` - The type of the token manager associated with the given token ID.
-    fn token_manager_type(env: &Env, token_id: BytesN<32>) -> TokenManagerType {
-        Self::token_id_config(env, token_id).token_manager_type
+    fn token_manager_type(
+        env: &Env,
+        token_id: BytesN<32>,
+    ) -> Result<TokenManagerType, ContractError> {
+        Self::token_id_config(env, token_id).map(|config| config.token_manager_type)
```
Comment (milapsheth) [inline contracts/interchain-token-service/src/contract.rs:173]:
```suggestion
    fn token_manager_type(
        env: &Env,
        token_id: BytesN<32>,
    ) -> TokenManagerType {
        Self::token_id_config(env, token_id).expect("token id config not found").token_manager_type
```
these entrypoints should still return the type as before

File: contracts/interchain-token-service/src/contract.rs [hunk 2]
```diff
@@ -509,11 +493,11 @@ impl InterchainTokenService {
                 token_handler::give_token(
                     env,
                     &destination_address,
-                    Self::token_id_config(env, token_id.clone()),
+                    Self::token_id_config(env, token_id.clone())?,
                     amount,
                 )?;
 
-                let token_address = Self::token_address(env, token_id.clone());
+                let token_address = Self::token_address(env, token_id.clone())?;
```
Comment (milapsheth) [inline contracts/interchain-token-service/src/contract.rs:500]: we're querying the token id twice. query it once, and retrieve the token address from within it

File: contracts/interchain-token-service/src/contract.rs (hunk 1 above)
Comment (ahramy) [reply contracts/interchain-token-service/src/contract.rs:173]: updated

File: contracts/interchain-token-service/src/contract.rs (hunk 2 above)
Comment (ahramy) [reply contracts/interchain-token-service/src/contract.rs:500]: yeah.. that's true. updated.

**Pull request merged**